contracts/Core/StakeManager.sol

```diff
@@ -292,17 +292,18 @@ contract StakeManager is Initializable, ACL, StakeStorage {
     }
 
     /// @notice The function is used by the Votemanager reveal function
-    /// to penalise the staker who lost his secret and make his stake zero and
-    /// transfer to bounty hunter half the schelling tokens of the stakers stake
+    /// to penalise the staker who lost his secret and 
+    /// to transfer half of penalty as "bountyReward" to bountyHunter
     /// @param id The ID of the staker who is penalised
     /// @param bountyHunter The address of the bounty hunter
     function slash (uint256 id, address bountyHunter, uint256 epoch) external onlyRole(parameters.getStakeModifierHash()) {
-        uint256 halfStake = stakers[id].stake/(2);
-        _setStakerStake(id, 0, "Slashed", epoch);
-        if (halfStake > 1) {
-            require(sch.transfer(bountyHunter, halfStake), "failed to transfer bounty");
-        }
-    } 
+      uint256 bountyReward = ((stakers[id].stake)*(parameters.percentSlashPenalty()))/(2*100);
+      uint256 Stake =  stakers[id].stake - ((stakers[id].stake*parameters.percentSlashPenalty())/100);
+      if (bountyReward > 1) {
```

Comment:
IMO We don't need this if, what it solves I am not clear
There are only two corner cases it excludes

bountyReward ==1 > I am not sure why we are not awarding bounty in this case
bountyReward ==0  > Even if its zero, sch.transfer wont transfer anything 

What do you guys say?
Maybe we should create diff issue and sep PR for same

Reply:
if (bountyReward > 0 && bountReward<1)
I think may be it is handling that @abhishekvispute

Reply:
I agree with @abhishekvispute here. I would also suggest we remove that if statement

Reply:
+1

Reply:
Recorded this in #162

Reply:
if (bountyReward = 0) return;
// go ahead

Reply:
will update this at once after updating all the test cases regarding this.